src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -336,164 +350,195 @@ public Mono<TransactionInfoDto> updateTransactionAuthorization(
+                                                                        TransactionUpdateAuthorizationCommand transactionUpdateAuthorizationCommand = new TransactionUpdateAuthorizationCommand(
+                                                                                transaction.getPaymentNotices().get(0)
+                                                                                        .rptId(),
+                                                                                updateAuthorizationStatusData
+                                                                        );
+
+                                                                        return transactionUpdateAuthorizationHandler
+                                                                                .handle(
+                                                                                        transactionUpdateAuthorizationCommand
+                                                                                )
+                                                                                .doOnNext(
+                                                                                        authorizationStatusUpdatedEvent -> log
+                                                                                                .info(
+                                                                                                        "Requested authorization update for rptId: {}",
+                                                                                                        transaction
+                                                                                                                .getPaymentNotices()
+                                                                                                                .get(0)
+                                                                                                                .rptId()
+                                                                                                )
+                                                                                )
+                                                                                .flatMap(
+                                                                                        authorizationStatusUpdatedEvent -> authorizationUpdateProjectionHandler
+                                                                                                .handle(
+                                                                                                        authorizationStatusUpdatedEvent
+                                                                                                )
+                                                                                );
+                                                                    }
```

Comment:
This branch is quite complex visually but semantically does one thing (updates a transaction authorization status, while the retrying mechanism through queues can be seen as an implementation detail of the aforementioned operation), it could be worth it to extract a separate function and call it here :)

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/168/commits/abfdb3480ce8cb9b2d53ad44070cbd6535e71c4c